src/playground-code-editor.ts

```diff
@@ -101,9 +111,36 @@ export class PlaygroundCodeEditor extends LitElement {
     playgroundStyles,
   ];
 
-  // Used by tests.
   protected _codemirror?: ReturnType<typeof CodeMirror>;
 
+  get cursorPosition(): Position | undefined {
+    const cm = this._codemirror;
+    if (!cm) return undefined;
+
+    return cm.getCursor('start');
+  }
```

Comment:
```suggestion
  get cursorPosition(): Position {
    return this._codemirror?.getCursor('start') ?? {ch: 0, line: 0};
  }
```

Reply:
There's also a `sticky` property in this codemirror `Position` type. It's not documented, do you know what it does? If not needed right now, maybe we should strip that (can always add it later, but seems like a codemirror-specific thing).